FIX(client): Windows loading/saving shortcuts

Septarius opened the pull request:
### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### Issues

This pull request fixes the following issues:

The revamped raw inputs are not saving / loading shortcuts for the Windows client.

Keyboard shortcuts that are outside of the normal range show up as "Unknown"

Fix the shortcuts events not firing for input buttons when the structs were compared using the default memcmp.

### Tested

- [x] InputXinput (Xbox controller)
- [x] InputKeyboard (Normal and "Unknown" characters)
- [x] InputMouse
- [x] InputHid ( Turn off XInput per @davidebeatrici Thanks!)
- [ ] InputGKey

I don't believe InputGKey is compatible with my mouse under Logitech GHub.

![image](https://user-images.githubusercontent.com/1136966/117616679-acbfef00-b128-11eb-9bd5-9329b1af2c1a.png)

1. Is the Xinput Xbox controller's A.
2. Is the InputHid Xbox controller's A.
3. M3 is for Mouse3.
4. K118 is "F24" on the keyboard which is mapped from mouse using Logitech GHub.


### Notes

This could be considered a breaking change since the shortcut save format might not be backwards compatible. Though that format change had already been made, this fixes it so the change persists to disk.
It is also difficult to reload a previous version's shortcuts to be sure what would be backwards compatible.

Krzmbrzl reviewed (changes requested):
I think your first commit should be of type `CODE` instead of `FIX`.

Krzmbrzl commented:
@davidebeatrici consider the changes made to `PluginManager.cpp` to be approved (thanks for finding that @Septarius :)).

davidebeatrici commented on src/mumble/GlobalShortcut_win.h:
```diff
@@ -30,6 +30,7 @@ class GlobalShortcutWin : public GlobalShortcutEngine {
 	///
 	/// @param msg The raw input handle to inject into GlobalShortcutWin.
 	void injectRawInputMessage(HRAWINPUT handle);
+	static void registerMetaTypes();
```
I would move this at the top and leave en empty line below:
```cpp
static void registerMetaTypes();

/// Inject a native Windows raw input message into GlobalShortcutWin's
/// thread. This method is meant to be called from the main thread
/// to pass native Windows keyboard messages to GlobalShortcutWin.
///
/// @param msg The raw input handle to inject into GlobalShortcutWin.
void injectRawInputMessage(HRAWINPUT handle);
```

davidebeatrici commented on src/mumble/GlobalShortcut_win.cpp:
```diff
@@ -635,6 +635,11 @@ GlobalShortcutWin::ButtonInfo GlobalShortcutWin::buttonInfo(const QVariant &butt
 		if (GetKeyNameText((input.code << 16) | (input.e0 << 24), buffer, MAX_PATH)) {
 			info.name = QString::fromWCharArray(buffer);
+		} else {
+			// If GetKeyNameText cannot find the name. Show a K prefix and the scan code instead of Unknown.
```
```suggestion
			// If GetKeyNameText() cannot find the name. Show a K prefix and the scan code instead of Unknown.
```

davidebeatrici commented:
> I don't have a controller that works under InputHid.

Xbox controllers are supported through raw input, you just have to disable XInput.

davidebeatrici commented:
Thanks for addressing my comments!

There was no need to set me as co-author for 11f96da31ce3155d27eff5ddc1678044180ceb72, I only suggested to move a function declaration after all.

Septarius commented:
Ok, I did add the additional commit to fix InputHid since I was able to test that one.

Krzmbrzl commented:
@Septarius I have to apologize: I advised you to use the `CODE` tag where I should have advised to use `REFAC` instead as there is no `CODE` type according to our [guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md) :see_no_evil:

Septarius commented:
No problem, fixed the commit summaries. Removed the errant co-author footers.
Format document didn't run on the last commit (1a04a17), so line 68 is now correct.
Thanks!
